tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -84,9 +84,9 @@ public IntelligentTestRunnerClient(string? workingDirectory, CIVisibilitySetting
         // Extract custom tests configurations from DD_TAGS
         _customConfigurations = GetCustomTestsConfigurations(_settings.TracerSettings.GlobalTags);
 
-        _getRepositoryUrlTask = GetRepositoryUrlAsync();
-        _getBranchNameTask = GetBranchNameAsync();
-        _getShaTask = GetCommitShaAsync();
```

Comment:
I guess the one difference here is that you're now running these in series when creating this type. Are you ok with that change? Would you rather defer calling each of those inside `UploadRepositoryChanges`? Is there any chance that you _won't_ have to run `Upload` (and so could avoid doing this relatively expensive work?)

e.g. it looks like you check each one for null/empty at the use site, and bail. So in the case where one fails, you could avoid doing the extra work?

May not be worth the extra complexity, just wondering 🙂 You _Could_ make them `Lazy<string>` for example as a simple "solution". Depends if you think it's worth it!

Reply:
Yes I thought about that and in the end for all intelligent test runner transactions we will need a git repo, so having these at .ctor level seems ok to me. 

Another thing is that we always upload the repository changes by default, we have a way to not upload but only as a kill switch. so 99.5% of case we will be running at least 1 transaction inside this class.